nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/StatusMerger.java

```diff
@@ -1015,4 +1019,39 @@ public static String prettyPrint(final Integer count, final Long bytes) {
         return formatCount(count) + " (" + formatDataSize(bytes) + ")";
     }
 
+    public static void merge(final ControllerServiceStatusDTO target, final ControllerServiceStatusDTO toMerge) {
+        if (target == null || toMerge == null) {
+            return;
+        }
+
+        // RunStatus for ControllerServiceStatusDTO can be one of [ENABLED, ENABLING, DISABLED, DISABLING]
+        if (ControllerServiceState.DISABLING.toString().equalsIgnoreCase(toMerge.getRunStatus())
+            && ControllerServiceState.DISABLED.toString().equalsIgnoreCase(target.getRunStatus())) {
+            target.setRunStatus(ControllerServiceState.DISABLING.toString());
+        }
+
+        if (ValidationStatus.VALIDATING.toString().equalsIgnoreCase(toMerge.getValidationStatus())) {
+            target.setValidationStatus(RunStatus.Validating.toString());
+        } else if (ValidationStatus.INVALID.toString().equalsIgnoreCase(toMerge.getRunStatus())) {
+            target.setValidationStatus(ValidationStatus.INVALID.toString());
+        }
+    }
+
+    public static void merge(final ReportingTaskStatusDTO target, final ReportingTaskStatusDTO toMerge) {
+        if (target == null || toMerge == null) {
+            return;
+        }
+
+        target.setActiveThreadCount(target.getActiveThreadCount() + toMerge.getActiveThreadCount());
+
+        // RunStatus for ReportingTaskStatusDTO can be one of [RUNNING, STOPPED, DISABLED]
+        target.setRunStatus(toMerge.getRunStatus());
+
+        if (ValidationStatus.VALIDATING.toString().equalsIgnoreCase(toMerge.getValidationStatus())) {
+            target.setValidationStatus(RunStatus.Validating.toString());
```

Comment:
It looks like the merged status be `ValidationStatus.VALIDATING.toString()` instead of `RunStatus.Validating.toString()`.
```suggestion
            target.setValidationStatus(ValidationStatus.VALIDATING.toString());
```